src/rise-watch.js

```
@@ -1,95 +1,63 @@
-    }`;
-
-    RisePlayerConfiguration.LocalStorage.watchSingleFile( filePath, _handleAttributeDataFileUpdateMessage );
+  function watchDataFile( filePath, handlerSuccess, handlerError ) {
+    RisePlayerConfiguration.LocalStorage.watchSingleFile( filePath, message => {
+      return _handleFileUpdateMessage( message, handlerSuccess, handlerError );
+    });
   }
 
-  function _handleAttributeDataFileUpdateMessage( message ) {
+  function _handleFileUpdateMessage( message, handlerSuccess, handlerError ) {
     if ( !message.status ) {
       return Promise.resolve();
     }
 
     switch ( message.status.toUpperCase()) {
-    case "FILE-ERROR":
-      return _handleAttributeDataFileUpdateError( message );
-
     case "CURRENT":
-      return _handleAttributeDataFileAvailable( message.fileUrl );
+      return _handleFileAvailable( message.fileUrl, handlerSuccess );
 
+    case "FILE-ERROR":
+      RisePlayerConfiguration.Logger.error(
+        WATCH_DATA_FILE, "data file error", message
+      );
+      // falls through
     case "NOEXIST":
     case "DELETED":
-      return RisePlayerConfiguration.AttributeData.sendStartEvent();
+      if ( handlerError ) {
+        return handlerError( message );
+      }
     }
 
     return Promise.resolve();
   }
 
-  function _handleAttributeDataFileUpdateError( message ) {
-    RisePlayerConfiguration.Logger.error(
-      WATCH_COMPONENT_DATA, "attribute data file error", message
-    );
-
-    return RisePlayerConfiguration.AttributeData.sendStartEvent();
-  }
-
-  function _handleAttributeDataFileAvailable( fileUrl ) {
+  function _handleFileAvailable( fileUrl, handlerSuccess ) {
 
     return RisePlayerConfiguration.Helpers.getLocalMessagingJsonContent( fileUrl )
       .then( data => {
-        return RisePlayerConfiguration.AttributeData.update( data );
+        return handlerSuccess( data );
       })
       .catch( error => {
         RisePlayerConfiguration.Logger.error(
```

@stulees @santiagonoguez one thing that I actually wanted to clarify. As you can see above, for the File Update message errors we call the error handler. However, for File Get errors we just log. I followed the same logic in my refactor, but I'm wondering if this was intentional or I should throw an error here as well. Thanks!